inttest/clusterreboot/clusterreboot_test.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022 k0s authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterreboot
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+
+	"github.com/k0sproject/k0s/inttest/common"
+)
+
+type ClusterRebootSuite struct {
+	common.FootlooseSuite
+}
+
+func (s *ClusterRebootSuite) TestK0sClusterReboot() {
+	s.T().Log("Starting k0s")
+	s.NoError(s.InitController(0))
+
+	kc, err := s.KubeClient(s.ControllerNode(0))
+	s.Require().NoError(err)
+
+	s.NoError(s.RunWorkers())
+	err = s.WaitForNodeReady(s.WorkerNode(0), kc)
+	s.NoError(err)
+
+	// reboot the cluster:
+	s.T().Log("Rebooting cluster")
+	s.rebootCluster()
+
+	// Restart k0s controller
+	s.T().Log("Restarting k0s")
+	s.NoError(s.InitController(0))
```

Comment:
If we use `LaunchMode: common.LaunchModeOpenRC` then we do not have to re-init controller, openRC should just start it on reboot automatically.